### PR TITLE
chore: reduce size of generated e2e buildspec

### PR DESCRIPTION
**Pull request by goldbez**
#### Description of changes
The purpose of this change is to reduce the size of the E2E buildspec file in order to meet the size requirement set by CodeBuild. This PR removes unnecessary fields such as "compute type", "image", and "type" (moving them to the job-specific buildspec file when necessary). It also removes a minimum line-width, which reduces unnecessary newlines. 

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild_specs/run_e2e_tests_windows.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 version: 0.2
 env:
   shell: powershell.exe
+  type: WINDOWS_SERVER_2019_CONTAINER
+  image: $WINDOWS_IMAGE_2019
```
Comment (awsluja) [inline codebuild_specs/run_e2e_tests_windows.yml:5]: nice

File: scripts/split-e2e-tests-codebuild.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export function loadConfigBase() {
   return yaml.load(fs.readFileSync(CODEBUILD_CONFIG_BASE_PATH, 'utf8'));
 }
 export function saveConfig(config: any): void {
-  const output = ['# auto generated file. DO NOT EDIT manually', yaml.dump(config, { noRefs: true })];
+  const output = ['# auto generated file. DO NOT EDIT manually', yaml.dump(config, { noRefs: true, lineWidth: -1 })];
```
Comment (awsluja) [inline scripts/split-e2e-tests-codebuild.ts:103]: what does this do?

Comment (goldbez) [reply scripts/split-e2e-tests-codebuild.ts:103]: This removes the minimum line width, which reduces the number of unnecessary newline characters.

**Review by kuhlmanp — DISMISSED** [review]

**Review by kuhlmanp — APPROVED** [review]

**Review by awsluja — APPROVED** [review]
nice!

**Pull request merged**